### PR TITLE
Sunbird AI Ugandan low resource language dataset

### PR DESCRIPTION
**Pull request by ak3ra**
Multi-way parallel text corpus of 5 key Ugandan languages for the task of machine translation.

### CODE REVIEW

File: datasets/sunbird/README.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+---
+YAML tags:
+- copy-paste the tags obtained with the tagging app: https://github.com/huggingface/datasets-tagging
```
Comment (lhoestq) [inline datasets/sunbird/README.md:3]: Feel free to use the tagging app to create the YAML tags of this dataset :)

File: datasets/sunbird/README.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+---
+
+# Dataset Card for [Sunbird AI Parallel Text MT Dataset]
```
Comment (lhoestq) [inline datasets/sunbird/README.md:6]:
```suggestion
# Dataset Card for Sunbird AI Parallel Text MT Dataset
```

File: datasets/sunbird/README.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+## Table of Contents
+- [Dataset Card for [Dataset Name]](#dataset-card-for-dataset-name)
```
Comment (lhoestq) [inline datasets/sunbird/README.md:9]:
```suggestion
- [Dataset Card for Sunbird](#dataset-card-for-sunbird)
```

File: datasets/sunbird/README.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    - [Citation Information](#citation-information)
+    - [Contributions](#contributions)
+
+## Dataset Description
+
+- **Homepage: sunbird.ai**
+- **Repository: https://github.com/SunbirdAI/ug-language-parallel-text-dataset**
+- **Paper:**
+- **Leaderboard:**
+- **Point of Contact: sunbird.ai**
```
Comment (lhoestq) [inline datasets/sunbird/README.md:44]:
```suggestion
- **Homepage:** sunbird.ai
- **Repository:** https://github.com/SunbirdAI/ug-language-parallel-text-dataset
- **Paper:**
- **Leaderboard:**
- **Point of Contact:** sunbird.ai
```

File: datasets/sunbird/README.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+en-run - English to Runyankore
+en-ach - English to Acholi
+en-teo - English to Itesot
+en-lgg - English to Lugbara
+
+## Dataset Structure
+
+### Data Instances
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/sunbird/README.md:67]: Could you add a sample from the dataset here please ?

You can get one by using `load_dataset` and taking the first sample `dataset["train"][0]`

File: datasets/sunbird/sunbird.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+import datasets
+
+
+_DESCRIPTION = """\
+Parallel train/test/eval dataset for low-resource machine translation Ugandan Languages.
+"""
+
+_CITATION = """\
+        TO ADD: Sunbird citation
```
Comment (lhoestq) [inline datasets/sunbird/sunbird.py:32]: don't forget to add the citation ;)

File: datasets/sunbird/sunbird.py [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        
+        lang_pair = self.config.language_pair
+        src_tag, tgt_tag = lang_pair.split("-")
+
+        domain = "supervised"
+
+        if domain=="supervised":
```
Comment (lhoestq) [inline datasets/sunbird/sunbird.py:93]: Is this really useful to do this ?

File: datasets/sunbird/sunbird.py [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+            print(_URL["supervised"].format(lang_pair))
```
Comment (lhoestq) [inline datasets/sunbird/sunbird.py:94]: Could you remove the print please ?

File: datasets/sunbird/sunbird.py [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+            dl_dir = dl_manager.download_and_extract(_URL["supervised"].format(lang_pair))
+        
+        data_dir = os.path.join(dl_dir, os.path.join("v1-dataset/v1.0", domain, lang_pair))
```
Comment (lhoestq) [inline datasets/sunbird/sunbird.py:97]:
```suggestion
        data_dir = os.path.join(dl_dir, "v1-dataset", "v1.0", domain, lang_pair)
```

File: datasets/sunbird/sunbird_template.py [hunk 10]
```diff
@@ -0,0 +1,176 @@
+# coding=utf-8
```
Comment (lhoestq) [inline datasets/sunbird/sunbird_template.py:1]: could you remove this file please ?

**Review by lhoestq — COMMENTED** [review]
Hi ! Thanks a lot for adding this one :)

I left a few comments

Also in order to be able to run our test suite on this dataset we also need to provide some dummy data. You can take a look here to see how to add them:

https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#automatically-add-code-metadata

**Comment by lhoestq** [comment]
Hi @ak3ra , have you had a chance to take my comments into account ?

Let me know if you have questions or if I can help :)

**Comment by ak3ra** [comment]
@lhoestq  Working on this, thanks for the detailed review :)

**Review by ak3ra — COMMENTED** [review]
Fixed issues, added citation, added sample data

**Review by lhoestq — COMMENTED** [review]
Hi ! Thanks for continuing to work on this dataset :) It's been some time ^^'

To have a way to test this dataset script properly, we also need dummy data: they can be generated automatically using the command
```
datasets-cli dummy_data ./datasets/sunbird --auto_generate --match_text_files "*"
```

I also added a few comments:

**Comment by albertvillanova** [comment]
Thanks for your contribution, @ak3ra. Are you still interested in adding this dataset?

We are removing the dataset scripts from this GitHub repo and moving them to the Hugging Face Hub: https://huggingface.co/datasets

We would suggest you create this dataset there. Please, feel free to tell us if you need some help.

**Pull request closed without merging**